Snowflake.Data/Client/SnowflakeDbConnection.cs

```diff
@@ -61,10 +69,15 @@ public SecureString Password
 
         public bool IsOpen()
         {
-            return _connectionState == ConnectionState.Open;
+            return _connectionState == ConnectionState.Open && SfSession != null;

```

Comment:
What is the case when state is open and session is null?

Reply:
Basically whenever IsOpen was used code actually looked into underlying SfSession so introducing this additional check is also a protection against NPE and is just a refactor